index.js

```diff
@@ -0,0 +1,100 @@
+const { print, parse } = require('graphql')
+
+function defaultCompare (a, b) {
+  if (a < b) {
+    return -1
+  }
+  return 1
+}
+
+function compareNodeByName (a, b) {
+  return defaultCompare(a.name.value, b.name.value)
+}
+
+function compareNodeByKind (a, b) {
+  const kindOrder = {
+    DirectiveDefinition: 1,
+    ScalarTypeDefinition: 2,
+    EnumTypeDefinition: 3,
+    InterfaceTypeDefinition: 4,
+    ObjectTypeDefinition: 5,
+    UnionTypeDefinition: 6,
+    InputObjectTypeDefinition: 7,
+    DirectiveExtension: 8,
+    ScalarTypeExtension: 9,
+    EnumTypeExtension: 10,
+    InterfaceTypeExtension: 11,
+    ObjectTypeExtension: 12,
+    UnionTypeExtension: 13,
+    InputObjectTypeExtension: 14,
+    SchemaDefinition: 15,
+    SchemaExtension: 16,
+    OperationDefinition: 17,
+    FragmentDefinition: 18
+  }
+  if (kindOrder[a.kind] < kindOrder[b.kind]) {
+    return -1
+  }
+  return 1
+}
+
+function sortDefinitions (definitions) {
+  return definitions.sort((a, b) => {
+    if (a.kind === b.kind) {
+      return compareNodeByName(a, b)
+    }
+    return compareNodeByKind(a, b)
+  })
+}
+
+function sortDefinitionNode (definition) {
+  return Object.entries(definition).reduce((acc, [key, value]) => {
+    if (key === 'fields' || key === 'directives' || key === 'arguments' || key === 'values' || key === 'types' || key === 'interfaces') {
+      acc[key] = value.map(sortDefinitionNode).sort(compareNodeByName)
+      return acc
+    }
+    if (key === 'locations') {
+      acc[key] = value.sort((a, b) => defaultCompare(a.value, b.value))
+      return acc
+    }
```

Comment:
Could we simplify this a teeny bit with a reversal of these blocks? If locations, use the default compare, otherwise, use compareNodeByName. On the flip side of that, we reduce how declarative the code is by doing that. If we want to keep it as is, should these strings be pulled into constants? Not sure if thats useful in this case.

Your call at any rate 👍 

```suggestion
    if (key === 'locations') {
      acc[key] = value.sort((a, b) => defaultCompare(a.value, b.value))
      return acc
    } else {
      acc[key] = value.map(sortDefinitionNode).sort(compareNodeByName)
      return acc
    }
   }

```

Reply:
These aren't the only keys in the entries that we're reducing though. They're just ones that correspond to the values we want sorted. Other keys with non-array values would throw an error.